### PR TITLE
fix: deal with X- entries

### PR DESCRIPTION
**Pull request by boredland**
closes https://github.com/Biont/sway-launcher-desktop/issues/25

### CODE REVIEW

**Comment by boredland** [comment]
hi @Biont could you have a look? code changes are minimal - most of it is tests.

**Comment by Biont** [comment]
@boredland Sorry for not replying earlier, I already reviewed it and it's a clean & well-made PR. Thank you very much for writing it.

What kept me from merging (and replying) is that I was planning to come back with a bit of research. 
What's bothering me a bit is that these `[X-* Shortcut Group]` seems to be deprecated as hell and - judging from google searches - seems to be an Ubuntu-only thing. Desktop actions were formalized [in version 1.1 of the spec](https://specifications.freedesktop.org/desktop-entry-spec/1.1/) which came out in April 2014. That's where [application actions are first mentioned](https://specifications.freedesktop.org/desktop-entry-spec/1.1/ar01s09.html)

The `TargetEnvironment=Unity` specifically -used by this Inkscape desktop file- is said to be deprecated even longer: These posts go back to 2012:
https://askubuntu.com/questions/232966/what-is-targetenvironment-unity
https://markmail.org/message/eypvnff4bieqf5bj

I realize that we're not talking about much code here. But deciding to support a spec that's deprecated since *at least* 6 years means we're also going to have to support its quirks. 
In other words: If I press merge here, I'm also subscribing for more work when implementing #34 because now we also have to support this other outdated stuff. And there might be more to come.

Do you have any insight in the importance of having this kind of support? How can we see how widespread the usage of these shortcut actions still is? If it's just one desktop file from Inkscape, I think it might be a better idea to urge the package maintainers to port the file to a more recent version of the specification

**Comment by boredland** [comment]
while I mostly agree, at least for inkscape that isn't an ubuntu issue (I'm using manjaro), but an inkscape issue:
https://gitlab.com/inkscape/inkscape/-/blob/master/org.inkscape.Inkscape.desktop.template

**Comment by Biont** [comment]
It's only a small change and if it ever acts up, it won't break anything when removed.
